engine/src/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java

```diff
@@ -188,7 +188,9 @@ private void executeTransformation() throws KettleException {
     }
     transExecutorData.groupTimeStart = System.currentTimeMillis();
 
-    discardLogLines( transExecutorData );
+    if ( first ) {
+      discardLogLines( transExecutorData );
+    }
```

Comment:
Missing other commit from TransExecutor around line 242 - final String logChannelId = executorTrans.getLogChannelId(); ... and about 8 other lines.